src/gui/editorwidgets/qgsvaluemapconfigdlg.cpp

```diff
@@ -175,16 +199,58 @@ void QgsValueMapConfigDlg::updateMap( const QList<QPair<QString, QVariant>> &lis
     ++row;
   }
 
+  constexpr int maxOverflowErrors { 5 };
+  QStringList reportedErrors;
+  const QgsField mappedField { layer()->fields().field( field() ) };
```

Comment:
Can we protect from fields() empty or field() equals to -1, even though is not suppose to happen

Reply:
done